### PR TITLE
[VL] Daily Update Velox Version (2024_03_05)

### PR DESCRIPTION
**Pull request by GlutenPerfBot**
Rebase velox 2024_03_05 
 
ccb464b06 by xiaoxmeng, Make the default 8-way join spill partitioning (#8951)
72413027c by Ge Gao, Fix flaky test AsyncDataCacheTest.ttl (#8946)
b11aeeb79 by joey.ljy, Allow choosing to register companion functions for all aggregate functions (#8855)
603554b22 by xumingming, Fix typo in Memory Management documentation (#8827)
3d9cf528b by Jimmy Lu, Table sampling push down
ffb7b0c82 by Nick Terrell, Switch lexing function to velox::Status and std::string_view
16a7fe109 by Deepak Majeti, Remove use of variant in SplitReader (#8929)
0fe23dfdb by Christian Zentgraf, Fix timestamp with timezone comparison (#8838)
0f881153e by duanmeng, Use ArbitratorTestUtil::createMemoryManager in SharedArbitrationTest (#8939)
d2f075fa4 by Christian Zentgraf, Remove unnecessary include of boost/locale.hpp (#8936)
7b4dc7be8 by Deepak Majeti, Fix CI Linux adapters job (#8942)
704d65d92 by Jimmy Lu, New exchange protocol (#8845)

### CODE REVIEW

File: .github/workflows/velox_be.yml [hunk 1]
```diff
@@ -436,8 +436,8 @@ jobs:
       - name: TPC-H SF1.0 && TPC-DS SF10.0 Parquet local spark3.2 with Celeborn
         run: |
           $PATH_TO_GLUTEN_TE/$OS_IMAGE_NAME/gha/gha-checkout/exec.sh \
-          'wget https://dlcdn.apache.org/incubator/celeborn/celeborn-0.3.0-incubating/apache-celeborn-0.3.0-incubating-bin.tgz && \
-          tar xzf apache-celeborn-0.3.0-incubating-bin.tgz && cd apache-celeborn-0.3.0-incubating-bin && \
+          'wget https://dlcdn.apache.org/incubator/celeborn/celeborn-0.3.2-incubating/apache-celeborn-0.3.2-incubating-bin.tgz && \
```
Comment (kerwin-zk) [inline .github/workflows/velox_be.yml:439]: Only need to replace the link of 0.3.0.  The new link is ```https://archive.apache.org/dist/incubator/celeborn/celeborn-0.3.0-incubating/apache-celeborn-0.3.0-incubating-bin.tgz```.  0.3.2 cannot be upgraded directly. Some APIs need to be adapted.

**Review by rui-mo — APPROVED** [review]

File: gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 2]
```diff
@@ -54,7 +54,9 @@ class VeloxTestSettings extends BackendTestSettings {
       "SPARK-32038: NormalizeFloatingNumbers should work on distinct aggregate",
       // Replaced with another test.
       "SPARK-19471: AggregationIterator does not initialize the generated result projection" +
-        " before using it"
+        " before using it",
+      // TODO: fix inconsistent behavior.
+      "SPARK-17641: collect functions should not collect null values"
```
Comment (rui-mo) [inline gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:59]: Could you paste the relevant issue here?

Comment (PHILO-HE) [reply gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:59]: https://github.com/oap-project/gluten/issues/4763

**Pull request merged**